odoo/addons/base/transpiler/transpiler_js.py

```diff
@@ -0,0 +1,393 @@
+        self.content = p.sub(repl, self.content)
+
+    def convert_relative_require(self):
+        """
+        Convert the relative require call that was generated by convert_unnamed_relative_import.
+        Simply give it the new path system.
+        Note: if the relative path uses the legacy system (addon.module) we keep it as it is valid.
+
+        before: require "some/path"
+        after: require "@module/path"
+        """
+        p = re.findall(r"""require\((["'])([^@\"\']+)(["'])\)""", self.content)
+        for open, path, close in p:
+            if not bool(re.match(r"\w+\.\w+", path)):
+                self.content = re.sub(rf"require\({str(open)}{path}{str(close)}\)", f'require("{self.relative_path_to_module_path(path)}")', self.content)
+
+    def convert_star_import(self):
+        """
+        Transpile import star.
+
+        before: import * as name from "some/path"
+        after: const name = require("some/path")
+        """
+        p = re.compile(r"^(?P<space>\s*)import\s+\*\s+as\s+(?P<identifier>\w+)\s*from\s*(?P<path>[^;\n]+)", re.MULTILINE)
+        repl = r"\g<space>const \g<identifier> = require(\g<path>)"
+        self.content = p.sub(repl, self.content)
+
+    def convert_unnamed_relative_import(self):
+        """
+        Transpile relative "direct" imports. Direct meaning they are not store in a variable.
+
+        before: import "some/path"
+        after: require("some/path")
+        """
+        p = re.compile(r"^(?P<space>\s*)import\s+(?P<path>[^;\n]+)", re.MULTILINE)
+        repl = r"require(\g<path>)"
+        self.content = p.sub(repl, self.content)
+
+    def remove_index(self):
+        """
+        Remove in the paths the /index.js.
+        We want to be able to import a module just trough its directory name if it contains an index.js.
+        So we no longer need to specify the index.js in the paths.
+        """
+        p = re.compile(r"require\s*\(\s*(?P<path>(\".*/index/?\")|('.*/index/?')|(`.*/index/?`))\s*\)", re.MULTILINE)
+        def repl(matchobj):
+            d = matchobj.groupdict()
+            path = d.get("path")
+            path = path[: path.rfind("/index")] + path[0]
+            return f"require({path})"
+        self.content = p.sub(repl, self.content)
+
+    def relative_path_to_module_path(self, path_rel):
+        """
+        Convert the relative path into a module path, which is more generic and fancy.
+
+        :param path_rel: a relative path to the current url. (self.url)
+        :return: module path (@module/...)
+        """
+        url_split = self.url.split("/")
+        path_rel_split = path_rel.split("/")
+        nb_back = len([v for v in path_rel_split if v == ".."]) + 1
+        result = "/".join(url_split[:-nb_back] + [v for v in path_rel_split if not v in ["..", "."]])
+        return self.url_to_module_path(result)
+
+    @staticmethod
+    def is_odoo_module(content):
```

Comment:
Should not be a static method. Is it even useful that this lives in this module at all? it seems to only be used in the JSAsset, should be a utility function over there.